### PR TITLE
Add Build managers and Update Build Querysets.

### PR DESCRIPTION
**Pull request by saadmk11**
This is a continuation of #5750.

### **Tasks:**

- [x] Add Build Model Managers
- [x] Update `Build` Querysets with `internal` and `external` managers everywhere.
- [x] Add tests for new `Managers`

### CODE REVIEW

**Comment by ericholscher** [comment]
@saadmk11 Is it possible to have this PR have a base of `saadmk11:version-update`? That way we can see the only the changes in this branch.

**Comment by stsewd** [comment]
@ericholscher github only allows to do that with branches in the same repo, not for forks. So, we could push the other branch to this repo to have that feature.

**Comment by ericholscher** [comment]
Pushed them up, you can see it here: https://github.com/rtfd/readthedocs.org/compare/version-update...build-manager-update

File: readthedocs/builds/querysets.py [hunk 1]
```diff
@@ -116,7 +116,6 @@ def api(self, user=None, detail=True):
 
 class BuildQuerySet(SettingsOverrideObject):
     _default_class = BuildQuerySetBase
-    _override_setting = 'BUILD_MANAGER'
```
[the remark this answers is not in the document]
Comment (saadmk11) [reply readthedocs/builds/querysets.py]: @ericholscher actually we are now using `BuildManager` not `BuildQueryset.as_manager()` so i put this settings in the BuildManager.

File: readthedocs/builds/views.py [hunk 2]
```diff
@@ -84,6 +84,21 @@ def post(self, request, project_slug):
 
 class BuildList(BuildBase, BuildTriggerMixin, ListView):
 
+    def get_queryset(self):
+        # this is used to include only internal version
+        # builds in the build list page
+        self.project_slug = self.kwargs.get('project_slug', None)
+        self.project = get_object_or_404(
+            Project.objects.protected(self.request.user),
```
Comment (ericholscher) [inline readthedocs/builds/views.py]: We shouldn't used `protected`, let's just use `public`.

**Comment by ericholscher** [comment]
Will merge this now, and we can fix up any issues later.

File: readthedocs/builds/views.py (hunk 2 above)
Comment (saadmk11) [reply readthedocs/builds/views.py]: @ericholscher This is default in the base class should we change this?

Comment (saadmk11) [reply readthedocs/builds/views.py]: @ericholscher This is default in the base class and we were using it should we change it?

**Pull request merged**